temporian/implementation/numpy/data/event.py

```diff
@@ -1,44 +1,37 @@
 from typing import Dict, List, Tuple, Any
 
 import numpy as np
-
+import pandas as pd
 from temporian.core.data.event import Event
 from temporian.core.data.feature import Feature
 from temporian.core.data import dtype
 from temporian.implementation.numpy.data.sampling import NumpySampling
 
 
 class NumpyFeature:
-    def __init__(
-        self, name: str, sampling: NumpySampling, data: np.ndarray
-    ) -> None:
+    def __init__(self, name: str, data: np.ndarray) -> None:
         if len(data.shape) > 1:
             raise ValueError(
                 "NumpyFeatures can only be created from flat arrays. Passed"
                 f" input's shape: {len(data.shape)}"
             )
-        if data.dtype.type not in DTYPE_MAPPING:
-            raise ValueError(
-                f"Unsupported dtype {data.dtype} for NumpyFeature. Supported"
-                f" dtypes: {DTYPE_MAPPING.keys()}"
-            )
+        if data.dtype.type is np.string_:
+            if data.dtype.type not in DTYPE_MAPPING:
+                raise ValueError(
+                    f"Unsupported dtype {data.dtype} for NumpyFeature."
+                    f" Supported dtypes: {DTYPE_MAPPING.keys()}"
+                )
 
         self.name = name
-        self.sampling = sampling
         self.data = data
         self.dtype = data.dtype.type
 
-    def schema(self) -> Feature:
-        return Feature(
-            name=self.name,
-            dtype=self.core_dtype(),
-            sampling=self.sampling.names,
-        )
-
     def __repr__(self) -> str:
         return f"{self.name}: {self.data.__repr__()}"
 
     def core_dtype(self) -> Any:
+        if self.dtype.type is np.string_:
+            return dtype.STRING
```

Comment:
See discussion above.
Short story, DTYPE_MAPPING index self.dtype, not self.dtype.type.
A string can have many different self.dtype values (notably, because there is a separate self.dtype for each maximum string length).